docs/selectors.md

```diff
@@ -1,20 +1,37 @@
+Selectors query elements on the web page for interactions, like [page.click](api.md#pageclickselector-options), and to obtain `ElementHandle` through [page.$](api.md#pageselector). Built-in selectors auto-pierce [shadow DOM](https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_shadow_DOM) roots.
 
 <!-- GEN:toc-top-level -->
-- [Selector syntax](#selector-syntax)
+- [Syntax](#syntax)
+- [Best practices](#best-practices)
 - [Examples](#examples)
-- [Built-in selector engines](#built-in-selector-engines)
+- [Selector engines](#selector-engines)
 <!-- GEN:stop -->
 
-## Selector syntax
+## Syntax
+Selectors are defined by selector engine name and selector body, `engine=body`. 
+
+* `engine` refers to one of the [supported engines](#selector-engines)
+  * Built-in selector engines: [css], [text], [xpath] and [id selectors][id]
+  * Learn more about [custom selector engines](./extensibility.md)
+* `body` refers to the query string for the respective engine
+  * For `text`, body is the text content
+  * For `css`, body is a [css selector](https://developer.mozilla.org/en/docs/Web/CSS/CSS_Selectors)
 
-Selector is a string that consists of one or more clauses separated by `>>` token, e.g. `clause1 >> clause2 >> clause3`.  When multiple clauses are present, next one is queried relative to the previous one's result.
+Body format is assumed to ignore leading and trailing white spaces, so that extra whitespace can be added for readability.
 
-Each clause contains a selector engine name and selector body, e.g. `engine=body`. Here `engine` is one of the supported engines (e.g. `css` or a custom one). Selector `body` follows the format of the particular engine, e.g. for `css` engine it should be a [css selector](https://developer.mozilla.org/en/docs/Web/CSS/CSS_Selectors). Body format is assumed to ignore leading and trailing white spaces, so that extra whitespace can be added for readability. If selector engine needs to include `>>` in the body, it should be escaped inside a string to not be confused with clause separator, e.g. `text="some >> text"`.
+### Short-forms
+For convenience, common selectors have short-forms:
+- Selector starting with `//` or `..` is assumed to be `xpath=selector`
+  - Example: `page.click('//html')` is converted to `page.click('xpath=//html')`.
+- Selector starting and ending with a quote (either `"` or `'`) is assumed to be `text=selector`
+  - Example: `page.click('"foo"')` is converted to `page.click('text="foo"')`.
+- Otherwise, selector is assumed to be `css=selector`
+  - Example: `page.click('div')` is converted to `page.click('css=div')`.
+
+### Combining selectors
+Selectors defined as `engine=body` can be combined with the `>>` token, e.g. `clause1 >> clause2 >> clause3`. When multiple clauses are present, next one is queried relative to the previous one's result.
```

Comment:
- Short-form selectors can be combined as well: `div >> ../../.. >> "text"`
- I'd replace "clause" with "selector" in this section.